app/services/invoices/void_service.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Invoices
+  class VoidService < BaseService
+    def initialize(invoice:)
+      @invoice = invoice
+      super
+    end
+
+    def call
+      return result.not_found_failure!(resource: 'invoice') if invoice.nil?
+
+      result.invoice = invoice
+
+      begin
+        invoice.void!
+
+        invoice.credits.each do |credit|
+          CreditNotes::RecreditService.new(credit:).call
```

Comment:
You can simply use the following form, as all services inherit from the BaseService, which has a `call`  class method that is just initializing an instance with all received parameters and calling the `call` method.

```suggestion
          CreditNotes::RecreditService.call(credit:)
```